File: src/main/java/com/trustline/trustline/config/exception/RestClientException.java
package com.trustline.trustline.config.exception;

import java.util.Collections;
import java.util.Map;

public class RestClientException extends AbstractException {
    public static final String REST_CLIENT_ERROR_CODE = "101";

    Map additionalInfo;

    public RestClientException(String message) {
        this(REST_CLIENT_ERROR_CODE, message);
        this.additionalInfo = Collections.emptyMap();
    }

    public RestClientException(String message, Map additionalInfo) {
        this(REST_CLIENT_ERROR_CODE, message);
        this.additionalInfo = additionalInfo;
    }

    public RestClientException(String code, String message) {
        super(code, message);
        this.additionalInfo = Collections.emptyMap();
    }

    public Map getAdditionalInfo() {
        return additionalInfo;
    }
}

File: src/main/kotlin/trustline/config/exception/RestClientException.kt
package trustline.config.exception;


class RestClientException : AbstractException {

    var additionalInfo: Map<String, Any> = emptyMap()
        private set

    constructor(message: String) : super(REST_CLIENT_ERROR_CODE, message) {
        additionalInfo = emptyMap()
    }

    constructor(message: String, additionalInfo: Map<String, Any>) : super(REST_CLIENT_ERROR_CODE, message) {
        this.additionalInfo = additionalInfo
    }

    constructor(code: String, message: String) : super(code, message) {
        additionalInfo = emptyMap()
    }

    companion object {
        const val REST_CLIENT_ERROR_CODE = "101"
    }
}
